Src/Particle/AMReX_Particle.H

```diff
@@ -188,6 +188,12 @@ struct ConstParticleCPUWrapper
     operator int () const noexcept { return static_cast<int>(m_idata & 0x00FFFFFF); }
 };
 
+std::uint64_t SetParticleIDandCPU (Long id, int cpu) {
+    std:uint64_t idcpu;
```

Comment:
to avoid read-before-write warning:
```suggestion
    std:uint64_t idcpu = 0;
```